src/utils/data.js

```diff
@@ -529,18 +529,34 @@ export class DataUtil {
     return durations;
   };
 
-  getTotalInsulinData = () => {
+  getTotalInsulinAndWeightData = () => { 
+    const weight = this.getWeight();
     const { basal, bolus } = this.getBasalBolusData();
-
+    
     const totalInsulin = _.reduce([basal, bolus], (result, value) => {
       const delivered = _.isNaN(value) ? 0 : value || 0;
       return result + delivered;
     }, 0);
 
     return {
       totalInsulin,
+      weight
     };
   };
+
+  getWeight = () => {
+    // retrieve lastest pump settings
+    const pumpSettings = _.last(this.filter.byType('pumpSettings').top(Infinity));
+    const parameters = _.get(pumpSettings, 'payload.parameters');
+    const weight = _.find(parameters, { 'name': 'WEIGHT' });
+
+    if (weight !== undefined) {
+      return weight;
+    }
+
+    return undefined;
+
+  };
```

Comment:
```suggestion
  /**
   * Retreive the patient weight from the pump settings
   * @returns {{ name: 'WEIGHT', value: string, unit: string, level: number }}
   */
  getWeight() {
    // retrieve lastest pump settings
    const pumpSettings = _.last(this.filter.byType('pumpSettings').top(Infinity));
    const parameters = _.get(pumpSettings, 'payload.parameters');
    const weight = _.find(parameters, { name: 'WEIGHT' });

    if (_.isEmpty(weight)) {
      return null;
    }

    return weight;
  }
```
Better to return `null` instead of `undefined`, because `typeof null === "object"`.
`undefined` is a Javascript aberration.
I generally prefer, when possible to use the 'positive' test, instead of the negative one.